### PR TITLE
Concordium client release October 2022

### PR DESCRIPTION
**Pull request by dg-concordium**
## Purpose

Inform users about update of Concordium Client.

## Changes

Release notes, downloads page, note added to import section of cc topic.

## Checklist

- [X] My code follows the style of this project.
- [X] The code compiles without warnings.
- [X] I have performed a self-review of the changes.
- [ ] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [ ] (If necessary) I have updated the CHANGELOG.

## CLA acceptance

_Remove if not applicable.

By submitting the contribution I accept the terms and conditions of the
Contributor License Agreement v1.0
- link: https://developers.concordium.com/CLAs/Contributor-License-Agreement-v1.0.pdf

- [X] I accept the above linked CLA.

### CODE REVIEW

File: source/mainnet/net/installation/downloads-testnet.rst [hunk 1]
```diff
@@ -93,10 +93,10 @@ The version of the Ledger App is the same for Mainnet and Testnet. So if you alr
 
 .. _concordium-node-and-client-download-testnet:
 
-Concordium Client v4.1.0
+Concordium Client v4.2.0
 ========================
 
--  `Download the Testnet Concordium Client for Linux <https://distribution.concordium.software/tools/linux/concordium-client_4.1.0>`_
+-  `Download the Testnet Concordium Client for Linux <https://distribution.concordium.software/tools/linux/concordium-client_4.2.0>`_
```
Comment (abizjak) [inline source/mainnet/net/installation/downloads-testnet.rst:99]:
```suggestion
-  `Download the Testnet Concordium Client for Linux <https://distribution.concordium.software/tools/linux/concordium-client_4.2.0-0>`_
```

File: source/mainnet/net/installation/downloads-testnet.rst [hunk 2]
```diff
@@ -110,13 +110,13 @@ Concordium Client v4.1.0
 
    where you replace `*` with |client-version|.
 
--  `Download the Testnet Concordium Client for macOS <https://distribution.concordium.software/tools/macos/signed/concordium-client-4.1.0.pkg>`_
+-  `Download the Testnet Concordium Client for macOS <https://distribution.concordium.software/tools/macos/signed/concordium-client-4.2.0.pkg>`_
```
Comment (abizjak) [inline source/mainnet/net/installation/downloads-testnet.rst:113]:
```suggestion
-  `Download the Testnet Concordium Client for macOS <https://distribution.concordium.software/tools/macos/signed/concordium-client-4.2.0-0.pkg>`_
```

File: source/mainnet/net/installation/downloads-testnet.rst [hunk 3]
```diff
@@ -110,13 +110,13 @@ Concordium Client v4.1.0
...
 
    - The macOS distribution is an installer that places an alias to the binary
      into the folder ``/usr/local/bin``. So after installing, you should have
      ``concordium-client`` on your path.
 
--  `Download the Testnet Concordium Client for Windows <https://distribution.concordium.software/tools/windows/signed/concordium-client_4.1.0.exe>`_
+-  `Download the Testnet Concordium Client for Windows <https://distribution.concordium.software/tools/windows/signed/concordium-client_4.2.0.exe>`_
```
Comment (abizjak) [inline source/mainnet/net/installation/downloads-testnet.rst:119]:
```suggestion
-  `Download the Testnet Concordium Client for Windows <https://distribution.concordium.software/tools/windows/signed/concordium-client_4.2.0-0.exe>`_
```

File: source/mainnet/net/installation/downloads.rst [hunk 4]
```diff
@@ -107,13 +107,13 @@ Concordium Client v4.1.0
 
    where you replace `*` with |client-version|.
 
--  `Download the Mainnet Concordium Client for macOS <https://distribution.concordium.software/tools/macos/signed/concordium-client-4.1.0.pkg>`_
+-  `Download the Mainnet Concordium Client for macOS <https://distribution.concordium.software/tools/macos/signed/concordium-client-4.2.0.pkg>`_
```
Comment (abizjak) [inline source/mainnet/net/installation/downloads.rst:110]:
```suggestion
-  `Download the Mainnet Concordium Client for macOS <https://distribution.concordium.software/tools/macos/signed/concordium-client-4.2.0-0.pkg>`_
```

File: source/mainnet/net/installation/downloads.rst [hunk 5]
```diff
@@ -107,13 +107,13 @@ Concordium Client v4.1.0
...
 
    - The macOS distribution is an installer that places an alias to the binary
      into the folder ``/usr/local/bin``. So after installing, you should have
      ``concordium-client`` on your path.
 
--  `Download the Mainnet Concordium Client for Windows <https://distribution.concordium.software/tools/windows/signed/concordium-client_4.1.0.exe>`_
+-  `Download the Mainnet Concordium Client for Windows <https://distribution.concordium.software/tools/windows/signed/concordium-client_4.2.0.exe>`_
```
Comment (abizjak) [inline source/mainnet/net/installation/downloads.rst:116]:
```suggestion
-  `Download the Mainnet Concordium Client for Windows <https://distribution.concordium.software/tools/windows/signed/concordium-client_4.2.0-0.exe>`_
```

File: source/mainnet/net/references/concordium-client.rst [hunk 6]
```diff
@@ -312,6 +312,10 @@ The ``--name`` option selects which account to import and imports it with this
 name. If it's omitted, all accounts in the file are imported under their
 existing names.
 
+.. Note::
+
+   When importing keys that have been exported from |bw|, ``--name`` must be included.
```
Comment (abizjak) [inline source/mainnet/net/references/concordium-client.rst:317]:
```suggestion
   When importing keys that have been exported from |bw|, the ``--name`` option must be provided to name the account.
```

File: source/mainnet/net/resources/release-notes-mainnet.rst [hunk 7]
```diff
@@ -12,6 +12,14 @@ Mainnet 4: Sirius
 
    Prior to Sirius, the nodes enforced that a transaction could not be deployed until 2 hours before its expiry date. With Sirius, node validation of transactions has been improved and the 2 hour window has been removed.
 
+October 5, 2022
+
+Concordium Client 4.2.0
+-----------------------
+
+- Fix handling of ``--no-confirm`` in ``contract init``, ``contract update``, ``module deploy``, and ``register data`` transactions. This flag is now respected.
+- Add support for import of keys from |bw|.
```
Comment (abizjak) [inline source/mainnet/net/resources/release-notes-mainnet.rst:21]: Did you deliberately omit the 
```
Make the output of CCD amounts consistent when printing transaction outcomes. The output should now always x.y CCD for some x and y.
```
?

Maybe we could instead add something like "Fix some inconsistencies in the display format of CCD amounts."?

File: source/mainnet/net/resources/release-notes.rst [hunk 8]
```diff
@@ -12,6 +12,14 @@ Sirius Testnet
 
    Prior to Sirius, the nodes enforced that a transaction could not be deployed until 2 hours before its expiry date. With Sirius, node validation of transactions has been improved and the 2 hour window has been removed.
 
+October 5, 2022
+
+Concordium Client 4.2.0
+-----------------------
+
+- Fix handling of ``--no-confirm`` in ``contract init``, ``contract update``, ``module deploy``, and ``register data`` transactions. This flag is now respected.
+- Add support for import of keys from |bw|.
```
Comment (abizjak) [inline source/mainnet/net/resources/release-notes.rst:21]: Same comment as for the testnet release notes.

**Review by abizjak — APPROVED** [review]

File: source/mainnet/net/resources/release-notes-mainnet.rst (hunk 7 above)
Comment (dg-concordium) [reply source/mainnet/net/resources/release-notes-mainnet.rst:21]: It wasn't in the changelog when I took a copy and made the pull request. So no, it was not deliberately omitted. I have added it now.

**Pull request merged**